### ISSUE TITLE
"Go to matching brace" not working as expected with macros.

### ISSUE THREAD

**Issue by Buggfix42** [notepad-plus-plus/notepad-plus-plus#13722]
### Description of the Issue
The "Go to matching brace" command seems to be actually split in two: Go from opening brace to closing brace and vice versa.
Depending on the brace type one of both is executed. Normally this isn't noticeable, except when you define a macro for it.

### Steps to Reproduce the Issue

1. Define a macro that uses the functionality "Go to matching brace"
2. Execution of that macro only works in one direction (opening brace to closing brace or vice versa)

### Debug Information
Reproduceable on v8.5.3

**Comment by alankilborn** [comment]
Probably need to elaborate your "Steps to Reproduce" with a real example.  From the limited description, it sounds to me as if it is working as intended.

**Comment by CennoxX** [comment]
1. This is a text with cursor: `|(example)`
2. Record a macro
3. Menu > Search > Go to Matching Brace
4. cursor changes position: `(example|)`
5. Stop macro
6. Set cursor at starting position: `|(example)`
7. Run macro
8. Expected result: `(example|)`, but the cursor doesn't change position
9. Set cursor at this position: `(example|)`
10. Run macro
11. Position of cursor changed as expected: `|(example)`

As said, the execution of that macro only works in one direction (opening brace to closing brace or vice versa)

**Comment by Buggfix42** [comment]
No. It depends on how you recorded the macro. If you recorded the macro by placing the cursor on a closing brace and execute "Go to matching brace" then the macro can only jump from a opening brace to an closing brace.
If you recorded the macro by placing the cursor on a opening brace and execute "Go to matching brace" then the macro can only jump from a closing brace to an opening brace.

**Comment by anatoly77g** [comment]
If you save the recorded macro, it appears in this file (Windows 11): %AppData%\Notepad++\shortcuts.xml
Looking in this file sheds some light on what is going on:
"Go to matching brace" is recorded in a macro in the same way as "Search -> Go to... -> Offset". That is, when playing back the macro, the cursor will jump to a fixed offset in the editing buffer, regardless of what text there is now.

So the problem is not that it jumps only forward or only backward.
The problem is that it jumps to a fixed position, regardless of the text currently being edited.

**Comment by victorel-petrovich** [comment]
@anatoly77g  
Perhaps this has changed a bit, but in N++ version 8.8.5, it records 2 actions, of which the first one is the one you mention and seems redundant, as detailed below: 
text file contents: 
```
 a(example) 

```
goMB_recStartBr=go to maching brace, recording started with cursor at start brace
```xml
        <Macro name="goMB_recStartBr" Ctrl="no" Alt="no" Shift="no" Key="0">
            <Action type="0" message="2025" wParam="9" lParam="0" sParam="" />
            <Action type="2" message="0" wParam="43009" lParam="0" sParam="" />
        </Macro>
```
goMB_recEndBr = go to maching brace, recording started with cursor at end brace
```xml
        <Macro name="goMB_recEndBr" Ctrl="no" Alt="no" Shift="no" Key="0">
            <Action type="0" message="2025" wParam="1" lParam="0" sParam="" />
            <Action type="2" message="0" wParam="43009" lParam="0" sParam="" />
        </Macro>
```
gotoOffs_1to9 = search > goto ... > offset, 1 to 9
```xml
        <Macro name="gotoOffs_1to9" Ctrl="no" Alt="no" Shift="no" Key="0">
            <Action type="0" message="2025" wParam="9" lParam="0" sParam="" />
        </Macro>
```
The type 2 action above is indeed the Menu> Search > Go to maching Brace, based on the ID found with NppUISpy plugin.

This bug is similar to https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5217#issuecomment-452717778

**Comment by victorel-petrovich** [comment]
This bug is similar to https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5217#issuecomment-452717778 
and so might be fixed in a similar way to https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14613
